.github/workflows/mdchecks.yml

```diff
@@ -0,0 +1,79 @@
+name: Markdown Checks
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths: '**.md'
+
+jobs:
+  mdchecks:
+    name: Markdown Checks
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout the Repo 
+      uses: actions/checkout@v2
+
+    - name: Set up NodeJS
+      uses: actions/setup-node@v1
+
+    - name: Get MD files in this PR
+      uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        # Paths to files will be available in `filter_files` output variable.
+        list-files: shell
+        filters: |
+          markdown:
+            - added|modified: '*.md'
+
+      # This step filters the list of files further, removing from consideration
+      # any file that includes the XML comment
+      # `<!-- BSSW_GITHUB_CI_IGNORE_MD_CHECKS -->` beginning on a first column
+      # anywhere in the file. The resulting list of files is available in
+      # `keep_files`.
+    - name: Skip Ignore Files
+      id: keep
+      run: |
+        mdfiles=""
+        for f in ${{ steps.filter.outputs.markdown_files }}; do 
+          if [ -n "$(grep '^<!-- BSSW_GITHUB_CI_IGNORE_MD_CHECKS -->' $f)" ]; then
+            continue
+          fi
+          mdfiles="$mdfiles $f"
+        done
+        echo ::set-output name=keep_files::${mdfiles}
+
+    - name: Install MD Tools
+      if: ${{ steps.keep.outputs.keep_files != '' }}
+      run: |
+        npm install -g markdownlint-cli
+        npm install -g markdown-spellcheck
+        npm install -g markdown-link-check
```

Comment:
Could we put this into a script called something like `install-markdown-checkers.sh` and then call this like:

```
      run: |
        <base-dir>/install-markdown-checkers.sh
```

That way, someone could run this locally to install the needed dependencies to be able to run these scripts locally.

Reply:
NOT: I installed these dependencies on my local Linux machine with npm and it takes a good bit of time.  My guess is that most of the time running this job will be just installing the dependencies in the VM.

Reply:
Logs from running the CI show that the install takes 5-10 secs. There is a way to effectively cache this too but haven't learned GHA to that depth yet.